pkg/telemetry/tracing.go

```diff
@@ -1,12 +1,53 @@
 package telemetry
 
 import (
+	"context"
+
+	"github.com/openfga/openfga/internal/build"
+	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
 	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
 )
 
-func NewNoopTracer() trace.Tracer {
-	return trace.NewNoopTracerProvider().Tracer("noop")
+const serviceName = "openfga"
+
+func MustNewTracerProvider(endpoint string, ratio float64) *sdktrace.TracerProvider {
+	res, err := resource.Merge(
+		resource.Default(),
+		resource.NewSchemaless(
+			semconv.ServiceNameKey.String(serviceName),
+			semconv.ServiceVersionKey.String(build.Version),
+		))
+	if err != nil {
+		panic(err)
+	}
+
+	exp, err := otlptracegrpc.New(context.Background(),
```

Comment:
Let's add a context with a timeout here so we don't potentially hang indefinitely on the `grpc.WithBlock()`.